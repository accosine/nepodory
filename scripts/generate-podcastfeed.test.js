import {
  addConfig,
  generateEpisodeItem,
  generatePodcastFeed,
} from "./generate-podcastfeed";

describe("generate-podcastfeed", () => {
  const singleEpisodeSetting = {
    bitlength: "0000000000",
    date: "Sun, 21 Feb 2021 19:15:00 GMT",
    duration: "00:13:37",
    episode: "https://example.com",
    guid: "0000000000000000000",
    number: "01",
    poster: "example.jpg",
    subtitle: "I am a subtitle",
    title: "I am a title",
    video: "example.mp4",
    content: "I am some content\n",
    configuration: {
      feedMarketingLinkMarkup: "Just some marketing bla bla",
      publisher: "ACME",
    },
  };
  const settings = {
    protocol: "https://",
    domain: "example.com",
    name: "Example",
    email: "hello@example.com",
    publisher: "ACME",
    description: "I am a description",
    language: "en",
    image: "example.jpg",
    keywords: "podcast",
    category: "category",
    subcategory1: "subcategory1",
    subcategory2: "subcategory2",
    year: new Date("June 12, 1929 01:00:00").getFullYear(),
    buildDate: new Date("June 12, 1929 01:00:00").toUTCString(),
    episodesWithConfig: [singleEpisodeSetting].map(addConfig),
  };

  it("adds environment variables to the config", () => {
    const feedMarketingLinkMarkup =
      "<p>The young are not afraid of telling the truth.</p>";
    const folderEpisodes = "mdx/episodes";
    const publisher = "ACME";
    const date = "Wed, 12 Jun 1929 00:00:00 GMT";
    process.env.FOLDER_EPISODES = folderEpisodes;
    process.env.PODCAST_MARKETING_LINKMARKUP = feedMarketingLinkMarkup;
    process.env.PODCAST_PUBLISHER = publisher;
    const config = addConfig({ date: "1929-06-12T01:00" });

    expect(config).toEqual({
      configuration: {
        feedMarketingLinkMarkup,
        publisher,
      },
      date,
    });
  });

  it("generated a piece of xml for the episode in the podcast feed", () => {
    const singleEpisode = generateEpisodeItem(singleEpisodeSetting);
    expect(singleEpisode).toMatchSnapshot();
  });

  it("generated a podcast xml feed", () => {
    expect(generatePodcastFeed(settings)).toMatchSnapshot();
  });
});
