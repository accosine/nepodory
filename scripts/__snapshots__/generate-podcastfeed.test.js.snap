// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate-podcastfeed generated a piece of xml for the episode in the podcast feed 1`] = `
"<item>
      <pubDate>Fri, 01 Jan 2021 20:15:00 GMT</pubDate>
      <title>I am a title</title>
      <description>I am some content</description>
      <content:encoded><![CDATA[<p>I am some content - <p>The young are not afraid of telling the truth.</p></p>]]></content:encoded>
      <guid>0000000000000000000</guid>
      <enclosure url=\\"https://example.com\\" type=\\"audio/mpeg\\" length=\\"0000000000\\"/>
      <itunes:duration xmlns:itunes=\\"http://www.itunes.com/dtds/podcast-1.0.dtd\\">00:13:37</itunes:duration>
      <itunes:author xmlns:itunes=\\"http://www.itunes.com/dtds/podcast-1.0.dtd\\">ACME</itunes:author>
      <itunes:subtitle xmlns:itunes=\\"http://www.itunes.com/dtds/podcast-1.0.dtd\\">I am a subtitle</itunes:subtitle>
    </item>
    "
`;

exports[`generate-podcastfeed generated a podcast xml feed 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<rss xmlns:googleplay=\\"http://www.google.com/schemas/play-podcasts/1.0\\" xmlns:itunes=\\"http://www.itunes.com/dtds/podcast-1.0.dtd\\" xmlns:dc=\\"http://purl.org/dc/elements/1.1/\\" xmlns:atom=\\"http://www.w3.org/2005/Atom\\" xmlns:content=\\"http://purl.org/rss/1.0/modules/content/\\" version=\\"2.0\\">
  <channel>
    <atom:link href=\\"https://://example.com/podcasts.xml\\" rel=\\"self\\" type=\\"application/rss+xml\\"/>
    <itunes:owner>
      <itunes:name>Example</itunes:name>
      <itunes:email>hello@example.com</itunes:email>
    </itunes:owner>
    <title>Example</title>
    <googleplay:author>ACME</googleplay:author>
    <description>I am a description</description>
    <itunes:image href=\\"example.jpg\\"/>
    <copyright>Â© 1929 ACME</copyright>
    <lastBuildDate>Wed, 12 Jun 1929 00:00:00 GMT</lastBuildDate>
    <pubDate>Wed, 12 Jun 1929 00:00:00 GMT</pubDate>
    <generator>next.js</generator>
    <language>en</language>
    <link>https://://example.com</link>
    <itunes:keywords xmlns:itunes=\\"http://www.itunes.com/dtds/podcast-1.0.dtd\\">podcast</itunes:keywords>
    <itunes:block>no</itunes:block>
    <itunes:author>ACME</itunes:author>
    <itunes:category text=\\"category\\"/>
    <itunes:category text=\\"category\\">
      <itunes:category text=\\"subcategory1\\"/>
    </itunes:category>
    <itunes:category text=\\"category\\">
      <itunes:category text=\\"subcategory2\\"/>
    </itunes:category>
    <itunes:block>no</itunes:block>
    <itunes:explicit>yes</itunes:explicit>
    <ttl>60</ttl>
    <item>
      <pubDate>Fri, 01 Jan 2021 20:15:00 GMT</pubDate>
      <title>I am a title</title>
      <description>I am some content</description>
      <content:encoded><![CDATA[<p>I am some content - </p>]]></content:encoded>
      <guid>0000000000000000000</guid>
      <enclosure url=\\"https://example.com\\" type=\\"audio/mpeg\\" length=\\"0000000000\\"/>
      <itunes:duration xmlns:itunes=\\"http://www.itunes.com/dtds/podcast-1.0.dtd\\">00:13:37</itunes:duration>
      <itunes:author xmlns:itunes=\\"http://www.itunes.com/dtds/podcast-1.0.dtd\\">undefined</itunes:author>
      <itunes:subtitle xmlns:itunes=\\"http://www.itunes.com/dtds/podcast-1.0.dtd\\">I am a subtitle</itunes:subtitle>
    </item>
    
  </channel>
</rss>"
`;
